internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c

```diff
@@ -322,6 +326,43 @@ int BPF_KPROBE(cap_capable)
     return 0;
 }
 
+static __always_inline u32 clear_mntns(u32 mntns) {
+    char comm[TASK_COMM_LEN] = {};
+    bpf_get_current_comm(comm, sizeof(comm));
+    trace_hook("clear_mntns mntns=%u comm=%s", mntns, comm);
+
+    for (int i = 0; i < sizeof(RUNC_DONE); i++) {
+        if (comm[i] != RUNC_DONE[i]) {
+            return 0;
+        }
+    }
+
+    trace_hook("clearing mount namespace: %u", mntns);
+
+    bpf_map_delete_elem(&mntns_syscalls, &mntns);
+    event_data_t * event = bpf_ringbuf_reserve(&events, sizeof(event_data_t), 0);
+    if (event) {
+        event->pid = bpf_get_current_pid_tgid() >> 32;
+        event->mntns = mntns;
+        event->type = EVENT_TYPE_CLEAR_MNTNS;
+        bpf_ringbuf_submit(event, 0);
+        return 0;
+    } else {
+        return -1;
+    }
+}
+
+SEC("tracepoint/sched/sched_prepare_exec")
+int sched_prepare_exec(struct trace_event_raw_sched_process_exec * ctx)
+{
+    bpf_printk("sched_prepare_exec");
+    struct task_struct * task = (struct task_struct *)bpf_get_current_task();
+    u32 mntns2 = BPF_CORE_READ(task, nsproxy, mnt_ns, ns.inum);
+    //clear_mntns(mntns2);
```

Comment:
Why is this commented out?

Reply:
Is it because you are trying the alternative solution?

Reply:
Yes, correct. `sched_prepare_exec` is very new (https://github.com/torvalds/linux/commit/c82389947d90c8b0dd206d9ef082191eb58df8fc), so I am experimenting with `sched_process_exit` instead. That doesn't fully work yet, I'm working on fixing it.